### ISSUE TITLE
Lazy constraints at root node?

### ISSUE THREAD

**Issue by CBongiova** [JuliaOpt/JuMP.jl#1226]
Hi, 

I have a MILP in which I have a subset of constraints are added as lazy constraints. I would like to know the value of the relaxed solution at the root node but I would then need to have the lazy constraints applied at the root node. I have already set addlazycallback(m, mylazygenerator,fractional=true) but this doesn't seem to add lazy constraints at the root node. 

I am using Gurobi as a solver, so in some sense I would like to set the attribute Lazy=3 
http://www.gurobi.com/documentation/7.5/refman/lazy.html#attr:Lazy

Is there a way to do this in JuMP? 
Thanks!

**Comment by odow** [comment]
You can access the Gurobi API directly. See below.
```julia
julia> using Gurobi, JuMP

julia> m=Model(solver=GurobiSolver())
Feasibility problem with:
 * 0 linear constraints
 * 0 variables
Solver is Gurobi

julia> @variable(m, x >= 0)
x

julia> c = @constraint(m, x >= 1)
x >= 1

julia> @variable(m, z, Bin)
z

julia> JuMP.build(m)

julia> grb = JuMP.internalmodel(m).inner
Gurobi Model:
    type   : LP (MIP)
    sense  : minimize
    number of variables             = 2
    number of linear constraints    = 1
    number of quadratic constraints = 0
    number of sos constraints       = 0
    number of non-zero coeffs       = 1
    number of non-zero qp objective terms  = 0
    number of non-zero qp constraint terms = 0

julia> Gurobi.set_intattrelement!(grb_c, "Lazy", c.idx, 3)

julia> Gurobi.update_model!(grb_c)

julia> Gurobi.get_intattrarray(grb_c, "Lazy", 1, 1)
1-element Array{Int32,1}:
 3
```

**Comment by CBongiova** [comment]
Otherwise, given that I just want to retrieve the value of the relaxed solution, after the insertion of lazy constraints, is it sufficient to set the following (in the informational callback)? 
addinfocallback(m, infocallback, when = :MIPNode)

**Comment by odow** [comment]
> Do I necessarily need to assign the constraints to a variable (in this case "c") to be called in Gurobi.set_intattrelement!(grb_c, "Lazy", c.idx, 3)

No. The `c.idx` argument to `set_intattrelement!` just refers to the 1-indexed (as opposed to the 0-index C API) row number of the constraint. `Gurobi.set_intattrelement!(grb_c, "Lazy", 1, 3)` is also valid and will set the `"Lazy"` attribute of the first constraint to `3`.

The other questions are not related to JuMP, and might be more appropriate for the [Gurobi mailing list](https://groups.google.com/forum/#!forum/gurobi), or the [Julia discourse](https://discourse.julialang.org/c/domain/opt).
